modules/document_repository/jsx/editForm.js

```diff
@@ -0,0 +1,251 @@
+              required={true}
+              disabled={true}
+              value={this.state.docData.forSite}
+            />
+            <SelectElement
+              name="instrument"
+              label="Instrument"
+              options={this.state.Data.instruments}
+              onUserInput={this.setFormData}
+              ref="instrument"
+              value={this.state.docData.instrument}
+            />
+            <TextboxElement
+              name="pscid"
+              label="PSCID"
+              onUserInput={this.setFormData}
+              ref="pscid"
+              disable = {true}
+              value={this.state.docData.pscid}
+            />
+            <TextboxElement
+              name="visitLabel"
+              label="Visit Label"
+              onUserInput={this.setFormData}
+              ref="visitLabel"
+              value={this.state.docData.visitLabel}
+            />
+            <TextareaElement
+              name="comments"
+              label="Comments"
+              onUserInput={this.setFormData}
+              ref="comments"
+              value={this.state.docData.comments}
+            />
+            <TextboxElement
+              name="version"
+              label="Version"
+              onUserInput={this.setFormData}
+              ref="version"
+              value={this.state.docData.version}
+            />
+          <ButtonElement label="Update File"/>
+        </FormElement>
+      </div>
+    );
+  }
+
+  /**
+   * Handles form submission
+   * @param {event} e - Form submition event
+   */
+  handleSubmit(e) {
+    e.preventDefault();
+
+    let formData = this.state.docData;
+    let formObject= new FormData();
+    for (let key in formData) {
+      if (formData[key] !== '') {
+        formObject.append(key, formData[key]);
+      }
+    }
+   fetch(this.props.action, {
+      method: 'POST',
+      cache: 'no-cache',
+      credentials: 'same-origin',
+      body: formObject,
+    })
+    .then((resp) => resp.json())
+    .then(()=>{
+      swal('Updated Successful!', '', 'success');
+      this.props.refreshPage();
+      this.fetchData();
+    });
+  }
+
+  /**
+   * Set the form data based on state values of child elements/componenets
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding form element
+   */
+  setFormData(formElement, value) {
+    let docData = this.state.docData;
+      docData[formElement] = value;
+    this.setState({
+      docData: docData,
+    });
+  }
+
+  /**
+   * Display a success/error alert message after form submission
+   */
+  showAlertMessage() {
+    let self = this;
+
+    if (this.refs['alert-message'] === null) {
+      return;
+    }
+
+    let alertMsg = this.refs['alert-message'];
+    $(alertMsg).fadeTo(2000, 500).delay(3000).slideUp(500, function() {
+      self.setState({
+        uploadResult: null,
+      });
+    });
+  }
+}
```

Comment:
This function is not called anywhere and should likely be removed and replaced with swals